### PR TITLE
Adding end-time fields and inserting

### PR DESCRIPTION
**Pull request by paul-christmann**
Not done yet, figured I'd get a PR created...

### CODE REVIEW

File: db/scripts/002_add_google_calendar.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE `qsignup`.`qsignups_regions`
+ADD COLUMN `google_calendar_id` VARCHAR(45) NULL AFTER `weekly_ao_reminders`;
```
Comment (evanpetzoldt) [inline db/scripts/002_add_google_calendar.sql:2]: As I mentioned in slack, there could be a desire to have `google_calendar_id` at the AO level... for example, our website has individual pages for each AO, and before we moved to QSignups we would have a calendar specific to that AO on each page. Maybe that's something to consider in a V2 of this.

File: db/tables/qsignups_weekly.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE `qsignups_weekly` (
+  `ao_channel_id` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  `event_day_of_week` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  `event_time` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  `event_type` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  `team_id` varchar(100) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  PRIMARY KEY (`ao_channel_id`,`event_day_of_week`,`event_time`,`team_id`)
```
Comment (evanpetzoldt) [inline db/tables/qsignups_weekly.sql:7]: Similar to comment above, at some point I may add `event_type` to primary key

File: development.txt [hunk 3]
```diff
@@ -0,0 +1,99 @@
+Instructions for creating a local development environment to build/test features.
```
Comment (evanpetzoldt) [inline development.txt:1]: Thanks for putting this together! I'm eager to try this on my own so I'm not always testing things on the production app 😆

File: app.py [hunk 4]
```diff
@@ -103,8 +110,12 @@ def get_user_names(array_of_user_ids, logger, client):
     return names
 
 def refresh_home_tab(client, user_id, logger, top_message, team_id, context):
-    print(client)
-    print(context)
+    print("CLIENT", client)
```
Comment (evanpetzoldt) [inline app.py:113]: Honestly, we can take a lot of these print statements out at some point... I put them in during early debugging, and was too lazy to take them back out. Some of them like client and context don't even return anything interesting

File: app.py [hunk 5]
```diff
@@ -970,14 +983,33 @@ def handle_manage_schedule_option_button(ack, body, client, logger, context):
...
                     "text": "Event Start",
                     "emoji": True
                 }
 		    },
+            {
+                "type": "input",
+                "block_id": "event_end_time_select",
+                "element": {
+                    "type": "timepicker",
+                    "initial_time": "06:30",
```
Comment (evanpetzoldt) [inline app.py:999]: Could make an argument for the default being 6:15, but that's nitpicky

**Review by evanpetzoldt — COMMENTED** [review]
This all looks great! I'd like to play around with my own dev environment before fully signing off and merging in, and that will need to be next week

File: db/tables/qsignups_master.sql [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  `event_date` date NOT NULL,
+  `event_time` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  `event_day_of_week` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  `event_type` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  `event_special` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci DEFAULT NULL,
+  `event_recurring` tinyint(1) NOT NULL,
+  `q_pax_id` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci DEFAULT NULL,
+  `q_pax_name` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci DEFAULT NULL,
+  `team_id` varchar(100) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  PRIMARY KEY (`ao_channel_id`,`event_date`,`event_time`,`team_id`)
```
[the remark this answers is not in the document]
Comment (paul-christmann) [reply db/tables/qsignups_master.sql:12]: Well, that's another conversation.  I generally move away from "logical" primary keys.  Every time I try to define them, I can almost guarantee that there will be an edge case somewhere else.

My suggestion is to use a sequence field as a primary key, and use that as the FK where needed.  Like the _master table could / should have a FK to the _weekly table (qsignups_weekly_id).  Don't rely on "logical" relationships ("only one workout at this time at this AO per day). 

If you want to talk about changing that kind of structure, there's a pretty straight forward path: add the fields to the db, run some scripts to set the FKs appropriately, and then update the code.

File: db/scripts/002_add_google_calendar.sql (hunk 1 above)
Comment (paul-christmann) [reply db/scripts/002_add_google_calendar.sql:2]: Probably worth hopping on a call to talk about what *is* an AO.  As you've gotten requests from other regions, are we clear about what that represents?  If so - answers like this (calendar IDs) may fall in to place.

File: app.py (hunk 5 above)
Comment (paul-christmann) [reply app.py:999]: Yeah, it should be.  I missed this one while I was digging through all of the constants.  That's part of future things I'd like to explore - getting rid of some duplications through the code, so I can only add this a single time.

**Pull request merged**